Tasks/Common/azure-arm-rest/azure-arm-endpoint.ts

```diff
@@ -28,12 +28,14 @@ export class AzureRMEndpoint {
             this.endpoint = {
                 subscriptionID: tl.getEndpointDataParameter(this._connectedServiceName, 'subscriptionid', true),
                 subscriptionName: tl.getEndpointDataParameter(this._connectedServiceName, 'subscriptionname', true),
-                servicePrincipalClientID: tl.getEndpointAuthorizationParameter(this._connectedServiceName, 'serviceprincipalid', false),
-                servicePrincipalKey: tl.getEndpointAuthorizationParameter(this._connectedServiceName, 'serviceprincipalkey', false),
+                servicePrincipalClientID: tl.getEndpointAuthorizationParameter(this._connectedServiceName, 'serviceprincipalid', true),
```

Comment:
Why this change?

Reply:
Initially we have only one case SPN so servicePrincipalClientID and servicePrincipalClientKey is compulsory field But due to introduction of MSI these two are optional.